crawler-cli/src/main/java/org/gbif/crawler/pipelines/dwca/DwCAToAvroCommandVerification.java

```diff
@@ -0,0 +1,144 @@
+package org.gbif.crawler.pipelines.dwca;
+
+import org.gbif.common.messaging.api.messages.DwcaValidationFinishedMessage;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.eclipse.jdt.internal.core.Assert;
+
+/**
+ * Verifies the configuration parameters like existence of input and output path, and enables with calculated configuration parameters helping DwCAToAvro Conversion
+ */
+public class DwCAToAvroCommandVerification {
+
+  private final DwCAToAvroConfiguration configuration;
+  private DwcaValidationFinishedMessage receivedMessage;
+
+  public static DwCAToAvroCommandVerification of(DwCAToAvroConfiguration configuration) {
+    return new DwCAToAvroCommandVerification(configuration);
+  }
+
+  private DwCAToAvroCommandVerification(DwCAToAvroConfiguration configuration) {
+    this.configuration = configuration;
+  }
+
+  public DwCAToAvroCommandVerification with(DwcaValidationFinishedMessage receivedMessage) {
+    this.receivedMessage = receivedMessage;
+    return this;
+  }
+
+  /**
+   * verifies parameters and return calculated configuration parameters
+   *
+   * @return Calculated Configuration Parameters
+   */
+  public DwCA2AvroConfigurationParameter verifyParametersAndGetResourceConfigurations() {
+    /*
+      checks existence of archiveRepository provided in the yaml config
+     */
+    String inputBasePath = configuration.archiveRepository.endsWith(File.separator)
```

Comment:
the following lines can be simplified to something like:
java.nio.file.Path dwcaFilePath = Paths.get(configuration.archiveRepository, receivedMessage.getDatasetUuid() + ".zip");
then you can check existence using something like: dwcaFilePath.toFile().exists()
If we validate that the file exists is enough, I don't thin that we need to validate both the archiveRepo and then the file

Reply:
removed the check for archiveRepo.

updated change: 
Not using path as i still need to derive 2 paths (archive extraction path and archive path here). i also need to confirm i have file separator in end while deriving it.